pkg/apis/eks.cattle.io/v1/types.go

```diff
@@ -52,26 +52,40 @@ type EKSClusterConfigSpec struct {
 }
 
 type EKSClusterConfigStatus struct {
-	Phase          string   `json:"phase"`
-	VirtualNetwork string   `json:"virtualNetwork"`
-	Subnets        []string `json:"subnets"`
-	SecurityGroups []string `json:"securityGroups"`
+	Phase                    string    `json:"phase"`
+	VirtualNetwork           string    `json:"virtualNetwork"`
+	Subnets                  []string  `json:"subnets"`
+	SecurityGroups           []string  `json:"securityGroups"`
+	ManagedLaunchTemplateID  string    `json:"managedLaunchTemplateID"`
+	TemplateVersionsToDelete []*string `json:"templateVersionsToDelete"`
```

Comment:
Can you elaborate on this a bit, why does it happen afterwards. Is there a chance for a race condition between when the new one is added and old one is deleted in checkAndUpdate ?

Reply:
If the launch template version is deleted before the node group goes into updating status, then EKS will not allow the node group to be updated. EKS will put the node group in a bad status and the only recovery is to delete the node group and create a new one (or visa versa).